storage/layerhandler.go

```diff
@@ -13,7 +13,9 @@ type LayerHandler interface {
 	// Layer if possible, or nil and an error when unsupported. This may
 	// directly serve the contents of the layer or issue a redirect to another
 	// URL hosting the content.
-	Resolve(layer Layer) (http.Handler, error)
+	// The LayerHandler must also accept an options map, although it may or may
+	// not use each provided option.
+	Resolve(layer Layer, options map[string]interface{}) (http.Handler, error)
```

Comment:
This doesn't make sense to pass the options through here or change this interface. Why not detect it in the returned handler and set the method there?